### PR TITLE
Contributors API service by Web Crawler

### PR DESCRIPTION
**Pull request by Hiroshi18**
## Description
On this issue was not possible to use web crawler because the github uses JS to generate the graph, so to get the data would require to use a web driver a new dependencies.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by filipetoyoshima — CHANGES_REQUESTED** [review]
image

I've runned:
```bash
find . -name "*.sqlite3" -type f -delete
export TOKEN='<my token>'
export NAME='filipetoyoshima'
docker-compose up --build
```

**Comment by Hiroshi18** [comment]
It must be called from the gateway

**Comment by Hiroshi18** [comment]
I believe none of the app's can be tested this way since the refactoring of the project

**Comment by RomuloSouza** [comment]
If the problem is the query on get method, you can simply add the following line at the beginning of the method:
Return Response('test')
This way you can call the post method through the django rest interface

**Review by filipetoyoshima — CHANGES_REQUESTED** [review]
image

**Comment by filipetoyoshima** [comment]
I would like to say that is a corner case problem, but no:

image

File: hubcare/metrics/commit_metrics/contributors/views.py [hunk 1]
```diff
@@ -11,40 +12,78 @@
...
+        differents_authors_object.save()
+        serializer = DifferentsAuthorsSerializers(differents_authors)
+
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def get_contributors(self, owner, repo):
         username = os.environ['NAME']
         token = os.environ['TOKEN']
         github_request = requests.get(
             'https://api.github.com/repos/' + owner + '/' + repo + '/commits',
```
Comment (RomuloSouza) [inline hubcare/metrics/commit_metrics/contributors/views.py]: I suppose you this url isn't correct. This request returns only the last 30 commits. You should use the tag per_page=100 to get the last 100 commits, and if this does not solve the problem of 4 different conttributors, you should verify if there are more pages.

**Review by RomuloSouza — CHANGES_REQUESTED** [review]

**Review by cjjcastro — CHANGES_REQUESTED** [review]
Please solve the conflicts

File: hubcare/metrics/commit_metrics/contributors/views.py (hunk 1 above)
Comment (Hiroshi18) [reply hubcare/metrics/commit_metrics/contributors/views.py]: Alright will see it

**Comment by Hiroshi18** [comment]
> Please solve the conflicts

Ok did it now

**Comment by Hiroshi18** [comment]
He is back for another round...

Found a bug in the activity rate will try to fix

DeepinScreenshot_select-area_20190610213817

**Comment by Hiroshi18** [comment]
Think it is fixed

**Comment by filipetoyoshima** [comment]
That was a long fight

**Review by filipetoyoshima — CHANGES_REQUESTED** [review]
And looks like the fight isn't over yet.
image

`nodejs/node` works finnely.
`filipetoyoshima/test` attacks again.

**Review by filipetoyoshima — APPROVED** [review]
Well... I really dunno explain what happened, but I've tested again and it worked.
Computer Science is not an exact science, surely it is more like witchcraft or sorcery.

It is ok and pretty fast.

**Pull request merged**